### PR TITLE
[runbook task] Merge 5.0 into master

### PR DESCRIPTION
**Pull request by radoslaw-orlowski-amido**

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -31,3 +31,30 @@ data "azurerm_key_vault" "idam" {
   name                = local.vault_name
   resource_group_name = "idam-${var.env}"
 }
+
+module "redis-cache" {
```
Comment (kremi) [inline infrastructure/main.tf:35]: @henrydobson can you please confirm if that module is needed here?

Comment (henrydobson) [reply infrastructure/main.tf:35]: It is not.

File: infrastructure/main.tf [hunk 2]
```diff
@@ -31,3 +31,30 @@ data "azurerm_key_vault" "idam" {
...
+  name         = "redis-port"
+  value        = module.redis-cache.redis_port
+  key_vault_id = data.azurerm_key_vault.idam.id
+}
+
+resource "azurerm_key_vault_secret" "redis_key" {
+  name         = "redis-key"
+  value        = module.redis-cache.access_key
+  key_vault_id = data.azurerm_key_vault.idam.id
+}
```
Comment (henrydobson) [inline infrastructure/main.tf:60]:
```suggestion
```

**Review by henrydobson — CHANGES_REQUESTED** [review]

File: Jenkinsfile_CNP [hunk 3]
```diff
@@ -48,9 +48,9 @@ withPipeline(type, product, component) {
                 Using PREVIEW_ENVIRONMENT_NAME: ${env.PREVIEW_ENVIRONMENT_NAME}
                 Using NONPROD_ENVIRONMENT_NAME: ${env.NONPROD_ENVIRONMENT_NAME}""".stripIndent()
     }
-
+
```
Comment (henrydobson) [inline Jenkinsfile_CNP:51]: This change is littered with 👾

**Review by henrydobson — CHANGES_REQUESTED** [review]

**Review by kremi — DISMISSED** [review]

File: Jenkinsfile_CNP (hunk 3 above)
Comment (henrydobson) [reply Jenkinsfile_CNP:51]: There's no diff really. The merge is just adding whitespace aka space invaders

Comment (radoslaw-orlowski-amido) [reply Jenkinsfile_CNP:51]: 🤦 I didn't get the reference :)

Comment (henrydobson) [inline Jenkinsfile_CNP:51]:
```suggestion
```

File: Jenkinsfile_CNP [hunk 4]
```diff
@@ -68,7 +67,7 @@ withPipeline(type, product, component) {
                 Using NONPROD_ENVIRONMENT_NAME: ${env.NONPROD_ENVIRONMENT_NAME}
                 Using IDAMAPI: ${env.IDAMAPI}""".stripIndent()
     }
-
+
```
Comment (henrydobson) [inline Jenkinsfile_CNP:70]:
```suggestion

```

**Review by henrydobson — APPROVED** [review]

**Pull request merged**